src/components/Profile/EditProfile.js

```diff
@@ -32,63 +31,46 @@ export default function EditProfile() {
         privateProfile: true
     });
 
-    const user = MockUser[0];
-    const userEmail = JSON.parse(localStorage.getItem("email"));
-    const [registrationError, setRegistrationError] = React.useState({
-        message: "Error, please try again later",
-        hasError: false
-    });
-    const [confirmPassword, setConfirmPassword] = React.useState({password: '', isEqualToPassword: false});
+    const newPassword = useRef('');
+    const userEmail = React.useMemo(() => JSON.parse(localStorage.getItem("email")), []);
+    const token =  React.useMemo(() => JSON.parse(localStorage.getItem("token")), []);
+
+    const [registrationError, setRegistrationError] = React.useState("");
+    const [confirmPassword, setConfirmPassword] = React.useState('');
```

Comment:
What is the use of confirmPassword here? It is being set but is not being used. Can you clarify its purpose please?

Reply:
good catch, will remove. The only place i use the confirmPAssword is in the function that validates if its the same password. No need to have a state for this. can create the const locally